src/handler/mod.rs

```diff
@@ -0,0 +1,111 @@
+/// This module contains most of the low-level commands
+/// and grub variable modifications
+use anyhow::{bail, Result};
+
+use std::process::Command;
+use std::str;
+
+/// reboots the system if boot_counter is greater than 0 or can be forced too
+pub fn handle_reboot(force: bool) -> Result<()> {
+    if !force {
+        if let Some(t) = get_boot_counter()? {
+            if t <= 0 {
+                bail!("countdown ended, check greenboot-rollback status")
+            };
+        }
+    }
```

Comment:
can't you compress it to:

`if !force && if let Some(t) = get_boot_counter()? && t <= 0 {`

Reply:
not exactly whats suggested..
but able to reduce the nested loop in : ee8bd08b11f22f930ca7e0e7a43733cbd28dd8b2